src/controllers/auth.controller.js

```diff
@@ -0,0 +1,268 @@
+
+  if (userData) {
+    throw ApiError.badRequest('Only for not authorized');
+  }
+
+  if (!email || !password) {
+    throw ApiError.badRequest('Not all the credentials are provided!');
+  }
+
+  const user = await userService.getByEmail(email);
+
+  if (!user) {
+    throw ApiError.notFound();
+  }
+
+  const isPasswordValid = await bcrypt.compare(password, user.password);
+
+  if (!isPasswordValid) {
+    throw ApiError.badRequest('The username and/or password is incorrect.');
+  }
+
+  if (user.activationToken) {
+    throw ApiError.badRequest('The user`s email is not verified!');
+  }
+
+  sendAuthentication(res, user);
+};
+
+const refresh = async (req, res) => {
+  const refreshToken = req.cookies.refreshToken || '';
+  const token = await tokenService.getByToken(refreshToken);
+
+  if (!refreshToken || !token) {
+    res.clearCookie('refreshToken');
+    throw ApiError.unauthorized();
+  }
+
+  const userData = jwtService.validateRefreshToken(refreshToken);
+
+  if (!userData) {
+    throw ApiError.badRequest('Token is expired or revoked!');
+  }
+
+  await sendAuthentication(res, userData);
+};
+
+const logout = async (req, res) => {
+  const { refreshToken } = req.cookies;
+  const userData = await jwtService.validateRefreshToken(refreshToken);
+
+  if (!refreshToken || !userData) {
+    throw ApiError.unauthorized();
+  }
+
+  await tokenService.remove(userData.id);
+  res.clearCookie('refreshToken');
+  res.sendStatus(204);
+};
+
+const sendResetPassword = async (req, res) => {
+  const { email, password } = req.body;
+  const { refreshToken } = req.cookies;
+
+  if (!email || !password) {
+    throw ApiError.badRequest('Not all credentials are provided!');
+  }
+
+  const user = await User.findOne({ where: { email } });
+
+  if (refreshToken) {
+    const token = await tokenService.getByToken(refreshToken);
+
+    if (token) {
+      throw ApiError.badRequest(
+        'Action restricted to unauthorized users only!',
+      );
+    }
+  }
+
+  if (!user) {
+    throw ApiError.notFound();
+  }
+
+  const resetToken = jwtService.generateAccessToken({ email, password });
+
+  user.activationToken = resetToken;
+  user.save();
+
+  emailService.sendResetPasswordEmail(user.email, resetToken);
+  res.sendStatus(200);
+};
+
+const resetPassword = async (req, res) => {
+  const { refreshToken } = req.cookies;
+  const { resetToken } = req.params;
+
+  if (refreshToken) {
+    const isValidRefreshToken = await tokenService.getByToken(refreshToken);
+
+    if (isValidRefreshToken) {
+      throw ApiError.badRequest('Only for not authorized!');
+    }
+  }
+
+  const user = await User.findOne({ where: { activationToken: resetToken } });
+  const userData = await jwtService.validateAccessToken(resetToken);
+
+  if (!user) {
+    throw ApiError.notFound();
+  }
+
+  if (!userData) {
+    throw ApiError.badRequest('Invalid reset token');
+  }
+
+  const hashedPassword = await bcrypt.hash(
+    userData.password,
+    +process.env.HASH_ROUNDS,
```

Comment:
Consider using `parseInt(process.env.HASH_ROUNDS, 10)` instead of the unary plus for converting `HASH_ROUNDS` to an integer. This ensures consistency and clarity in your code.